src/coordination/coordinator_log_store.cpp

```diff
@@ -13,229 +13,164 @@
 
-  size_t accum_size = 0;
-  for (ulong ii = start; ii < end; ++ii) {
+  for (uint64_t i = start, curr_index = 0; i < end; ++i, ++curr_index) {
     ptr<log_entry> src = nullptr;
     {
-      std::lock_guard<std::mutex> l(logs_lock_);
-      auto entry = logs_.find(ii);
-      if (entry == logs_.end()) {
-        entry = logs_.find(0);
-        assert(0);
+      auto lock = std::lock_guard{logs_lock_};
+      if (auto const entry = logs_.find(i); entry != logs_.end()) {
+        src = entry->second;
+      } else {
+        throw RaftCouldNotFindEntryException("Could not find entry at index {}", i);
       }
-      src = entry->second;
     }
-    ret->push_back(MakeClone(src));
-    accum_size += src->get_buf().size();
-    if (batch_size_hint_in_bytes && accum_size >= (ulong)batch_size_hint_in_bytes) break;
+    (*ret)[curr_index] = MakeClone(src);
   }
   return ret;
 }
 
-ptr<log_entry> CoordinatorLogStore::entry_at(ulong index) {
+ptr<log_entry> CoordinatorLogStore::entry_at(uint64_t index) {
   ptr<log_entry> src = nullptr;
   {
-    std::lock_guard<std::mutex> l(logs_lock_);
-    auto entry = logs_.find(index);
-    if (entry == logs_.end()) {
-      entry = logs_.find(0);
-    }
-    src = entry->second;
+    auto lock = std::lock_guard{logs_lock_};
+    src = FindOrDefault_(index);
   }
   return MakeClone(src);
 }
 
-ulong CoordinatorLogStore::term_at(ulong index) {
-  ulong term = 0;
+uint64_t CoordinatorLogStore::term_at(uint64_t index) {
+  uint64_t term = 0;
   {
-    std::lock_guard<std::mutex> l(logs_lock_);
-    auto entry = logs_.find(index);
-    if (entry == logs_.end()) {
-      entry = logs_.find(0);
-    }
-    term = entry->second->get_term();
+    auto lock = std::lock_guard{logs_lock_};
+    term = FindOrDefault_(index)->get_term();
   }
   return term;
 }
 
-ptr<buffer> CoordinatorLogStore::pack(ulong index, int32 cnt) {
+ptr<buffer> CoordinatorLogStore::pack(uint64_t index, int32 cnt) {
   std::vector<ptr<buffer>> logs;
 
   size_t size_total = 0;
-  for (ulong ii = index; ii < index + cnt; ++ii) {
+  uint64_t const end_index = index + cnt;
+  for (uint64_t i = index; i < end_index; ++i) {
     ptr<log_entry> le = nullptr;
     {
-      std::lock_guard<std::mutex> l(logs_lock_);
-      le = logs_[ii];
+      auto lock = std::lock_guard{logs_lock_};
+      le = logs_[i];
     }
-    assert(le.get());
-    ptr<buffer> buf = le->serialize();
+    MG_ASSERT(le.get(), "Could not find log entry at index {}", i);
+    auto buf = le->serialize();
     size_total += buf->size();
     logs.push_back(buf);
   }
 
-  ptr<buffer> buf_out = buffer::alloc(sizeof(int32) + cnt * sizeof(int32) + size_total);
+  auto buf_out = buffer::alloc(sizeof(int32) + cnt * sizeof(int32) + size_total);
   buf_out->pos(0);
   buf_out->put((int32)cnt);
 
   for (auto &entry : logs) {
-    ptr<buffer> &bb = entry;
-    buf_out->put((int32)bb->size());
-    buf_out->put(*bb);
+    buf_out->put(static_cast<int32>(entry->size()));
```

Comment:
Do we have to do static cast from unsigned to signed?

Reply:
To avoid warning